urllib3/connectionpool.py

```diff
@@ -67,14 +67,26 @@ def __init__(self, host, port=None):
         if not host:
             raise LocationValueError("No host specified.")
 
+        self.host = self._ipv6_host(host)
+        self.port = port
+
+    def _ipv6_host(self, host):
+        """
+        Process IPv6 address literals
+        """
+
         # httplib doesn't like it when we include brackets in ipv6 addresses
         # Specifically, if we include brackets but also pass the port then
         # httplib crazily doubles up the square brackets on the Host header.
         # Instead, we need to make sure we never pass ``None`` as the port.
         # However, for backward compatibility reasons we can't actually
-        # *assert* that.
-        self.host = host.strip('[]')
-        self.port = port
+        # *assert* that.  See http://bugs.python.org/issue28539
+        #
+        # Also if an IPv6 address literal has a zone identifier, the
+        # percent sign might be URIencoded, convert it back into ASCII
+        if host[0] == '[' and host[-1] == ']':
+            host = host.replace('%25', '%').strip('[]')
+        return host
```

Comment:
Argh, sorry, should have noticed this earlier, but this method doesn't actually use `self` anywhere. That's usually a good sign that this is a function pretending to be a method, so we should hoist it off the class.